pandas/core/groupby/grouper.py

```diff
@@ -287,8 +290,11 @@ def __init__(
 
     @final
     @property
-    def ax(self):
-        return self.grouper
+    def ax(self) -> Index:
+        index = self._gpr_index
+        if index is None:
+            raise ValueError("_set_grouper must be called before ax is accessed")
```

Comment:
this hit in tests?

Reply:
no, in practice we always call _set_grouper before accessing this